### PR TITLE
Upgrade phpcodesniffer-composer-installer

### LINKED ISSUE #36913: [Issue] Upgrade phpcodesniffer-composer-installer
This issue is automatically created based on existing pull request: magento/magento2#36791: Upgrade phpcodesniffer-composer-installer

---------
### Description
`dealerdirect/phpcodesniffer-composer-installer` has now reached its first stable version. See https://github.com/PHPCSStandards/composer-installer/releases/tag/v1.0.0 for details.

### Manual testing scenarios
There are no functional changes between these two versions.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by fredden**
### Description
`dealerdirect/phpcodesniffer-composer-installer` has now reached its first stable version. See https://github.com/PHPCSStandards/composer-installer/releases/tag/v1.0.0 for details.

### Manual testing scenarios
There are no functional changes between these two versions.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#36913: Upgrade phpcodesniffer-composer-installer

### CODE REVIEW

**Comment by fredden** [comment]
@magento run all tests

File: composer.json
Comment (Den4ik) [inline composer.json:96]: As same as for codding-standard repo PR I believe that `"^1.0"` will be enough

**Review by Den4ik — COMMENTED** [review]
Hi @fredden Thanks for your update.
Please review my comment

File: composer.json
Comment (fredden) [reply composer.json:96]: There's nothing here that's *not* compatible with `^0.7.0`. I've specifically kept this because I expect other packages may take a while to upgrade / allow the new version. If there's something that requires `^0.7`, it'll work today, but not after merging if we change this to `^1.0` only. Listing both supported versions means that interoperability with other packages is maintained; Composer will pick the "best" version, which is typically the highest mutually-supported version. If we use `^1.0` only here, then we're forcing others to upgrade their packages in order for their coding standard to be used in conjunction with this product.

Comment (Den4ik) [reply composer.json:96]: Make sense

**Review by Den4ik — APPROVED** [review]
Failed test is not related to changes

**Comment by glo30253** [comment]
@magento run Functional Tests B2B

**Comment by glo30253** [comment]
Hi @fredden,
Thanks for the collaboration & contribution!

:heavy_check_mark: QA Passed

**Manual Scenario Steps**
**Before:** :heavy_multiplication_x: 
 We have checked with 2.4-develop and phpcodesniffer was 0.72 version
Screenshot 2023-02-15 at 11 30 40 AM
 
**After:** :heavy_check_mark:
We have checked with PR changes and phpcodesniffer composer installer upgrading from 0.72 t0 1.0
Screenshot 2023-02-15 at 11 32 54 AM
Hence moving it to merge in progress.

**Comment by engcom-Charlie** [comment]
As per above [comment](https://github.com/magento/magento2/pull/36791#issuecomment-1430803008), moving it to Merge In Progress.

Thank you!

**Comment by engcom-Lima** [comment]
@magento create issue

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
It looks like @glo71317 changed the version constraint from "^0.7" to "^1.0" in https://github.com/magento/magento2/commit/d337684914a45361d4a7ebc20e99faac97651239 (which is not associated with any pull request here). That's a breaking change. (See discussion here: https://github.com/magento/magento2/pull/36791#discussion_r1097133332.)

**Comment by fredden** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @fredden,

Thank you for your efforts and contribution!

As part of process, we were occupied with some priority PRs. The Communty voted PRs gets the priority which ultimately hold this PR back to move further. Meanwhile internal team worked on some feature which seems added these composer changes. We will see it further once we get some input from @glo71317 now.

@glo71317 can you please provide some information on [this](https://github.com/magento/magento2/pull/36791#issuecomment-1706333934).

Thank you!

**Comment by glo71317** [comment]
@fredden @Indraniks We received request [2.4.7-beta2] Platform Components Upgrade, In scope of the we have updated version constraint from "^0.7" to "^1.0" for **"dealerdirect/phpcodesniffer-composer-installer"** along with many more components. 
It is regular activity which we are performing before any release in scope of platform health. We are making sure all third dependencies which are using by adobe commerce that should be up to date.

**Comment by fredden** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Moving this PR back to Extended testing as few tests are failing and build is not green. 

Thank you!

**Comment by engcom-Charlie** [comment]
Hi @fredden,

Based on [this](https://github.com/magento/magento2/pull/36791#issuecomment-1709945338) comment, can we close this PR.

Thank you!

**Comment by fredden** [comment]
@engcom-Charlie, Adobe needs to decide if they want to accept the breaking changes introduced by @glo71317 in d337684914a45361d4a7ebc20e99faac97651239 (in which case this pull request should be closed), or not (in which case this pull request should be merged as was the plan since 15th February 2023).

**Comment by glo71317** [comment]
> @engcom-Charlie, Adobe needs to decide if they want to accept the breaking changes introduced by @glo71317 in [d337684](https://github.com/magento/magento2/commit/d337684914a45361d4a7ebc20e99faac97651239) (in which case this pull request should be closed), or not (in which case this pull request should be merged as was the plan since 15th February 2023).

Hi @fredden, i don't think there is any breaking changes, And also not forcing to any other vendor and customer to update the dependency in current magento version which they are using they can continue without any breaking changes, As per my understanding If they will upgrade with current version to upcoming 2.4.7-beta2 they can do without any breaking changes.
If you are facing any issue please let me know what are the breaking changes introduced by [d337684](https://github.com/magento/magento2/commit/d337684914a45361d4a7ebc20e99faac97651239)

**Comment by fredden** [comment]
> please let me know what are the breaking changes introduced by [d337684](https://github.com/magento/magento2/commit/d337684914a45361d4a7ebc20e99faac97651239)

I have already articulated this. Please see https://github.com/magento/magento2/pull/36791#issuecomment-1706333934
> That's a breaking change. (See discussion here: https://github.com/magento/magento2/pull/36791#discussion_r1097133332.)

**Comment by fredden** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@glo30253 I'm agree with @fredden 
We can't guarantee that thouthand's of Magento modules will work with this update. in this case Magento upgrade could break the stores. Make sense support both versions in the next minor release and remove old version at the second minor release. It provide some time to developers for update.

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@glo71317 you introduced the breaking change in d337684914a45361d4a7ebc20e99faac97651239. This pull request fixes the breaking part of that commit by re-introducing support for versions `0.7.x` of this package. Before your change, this pull request was already backwards-compatible. Please can you arrange for this to be merged, as was the intention back in February 2023. (See https://github.com/magento/magento2/pull/36791#event-8522492803)

**Comment by Den4ik** [comment]
@glo71317 @fredden provided enough comment

**Comment by glo71317** [comment]
@fredden @Den4ik Okay make sense thanks for your contribution.
@Indraniks Can you proceed for merging this pr?

**Comment by Den4ik** [comment]
@glo71317 Great, thanks for conversation

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests  EE, WebAPI Tests

**Comment by engcom-Charlie** [comment]
As the [this](https://github.com/magento/magento2/pull/36791#issuecomment-1911624690) comment, as well as the PR as already been tested prior moving this PR to Merge in Progress.

**Comment by Indraniks** [comment]
This PR got merged in [a2519ca](https://github.com/magento-commerce/magento2ce/commit/a2519ca33d89b974d929cf3ffcee142135e1f00e) hence closing this now.

**Comment by engcom-Charlie** [comment]
This PR got merged in [a2519ca](https://github.com/magento-commerce/magento2ce/commit/a2519ca33d89b974d929cf3ffcee142135e1f00e) hence closing this now.

**Comment by hostep** [comment]
@engcom-Charlie: please link to the publicly visible repo, it should be this link: https://github.com/magento/magento2/commit/a2519ca33d89b974d929cf3ffcee142135e1f00e

**Pull request closed without merging**
